machine-learning/04-openpbta_plot_LV.Rmd

```diff
@@ -0,0 +1,379 @@
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir, recursive = TRUE)
+}
+```
+
+#### Input
+
+```{r input_files}
+histologies_file <- file.path(data_dir, "pbta-histologies-stranded-rnaseq.tsv")
+plier_file <- file.path(models_dir, "pbta-medulloblastoma-plier.RDS")
+```
+
+#### Output
+
+```{r output_file}
+boxplot_png <- file.path(plots_dir, "medulloblastoma_LV57_boxplot.png")
+```
+
+## Read in data and prepare for plotting
+
+### Sample metadata
+
+```{r read_in_metadata}
+histologies_df <- read_tsv(histologies_file)
+```
+
+### PLIER results
+
+```{r read_in_plier, live = TRUE}
+# Read in the PLIER results
+plier_results <- read_rds(plier_file)
+```
+
+For biological discovery, we are often most interested in the latent variables that have some kind of association with an input gene set or pathway.
+We can use the FDR values in the summary data frame to filter to only the latent variables (and their associated gene sets).
+
+```{r filter_fdr, live = TRUE}
+# Filter to LV-pathway relationships with FDR < 0.05
+sig_summary_df <- plier_results$summary %>%
+  dplyr::filter(FDR < 0.05)
+sig_summary_df
+```
+
+```{r sig_index}
+# We only want a single instance of each LV index
+sig_index <- as.integer(unique(sig_summary_df$`LV index`))
+```
+
+```{r sig_b_wide, live = TRUE}
+# Get the LV by sample matrix from the PLIER results and subset it to only those
+# LVs with an FDR < 0.05 (at least one pathway)
+b_matrix <- plier_results$B
+sig_b_matrix <- b_matrix[sig_index, ]
+
+# Create a data frame and add a column with LV identifiers
+sig_b_wide <- data.frame(sig_b_matrix) %>%
+  tibble::rownames_to_column(var = "LV")
+```
+
+#### Tidy
+
+Let's look at what format this is in currently.
+
+```{r wide_peek}
+sig_b_wide
+```
+
+We want this in long format for plotting.
+We'll use `reshape2::melt` to do this just like in the toy example above.
+
+```{r melt_b, live = TRUE}
+sig_b_df <- reshape2::melt(sig_b_wide, 
+                           id.vars = "LV", 
+                           variable.name = "Kids_First_Biospecimen_ID",
+                           value.name = "LV_estimate")
+head(sig_b_df)
+```
+
+Now let's add relevant metadata to the data frame so we can use that for plotting.
+
+```{r add_metadata_b}
+sig_b_df <- inner_join(x = sig_b_df, 
+                       y = histologies_df %>%
+                         select(Kids_First_Biospecimen_ID,
+                                broad_histology,
+                                short_histology,
+                                integrated_diagnosis,
+                                molecular_subtype,
+                                composition),
+                       by = "Kids_First_Biospecimen_ID")
+```
+
+## Plotting
+
+We'll plot LV57; this is the latent variable that we looked at the loadings for in the previous notebook.
+
+```{r lv_to_plot}
+# PLIER names certain latent variables based on their association with input
+# gene sets
+lv_to_plot <- "57,SVM Macrophages M2"
```

Comment:
Sometimes! In this case yes because we've filtered to the significant association.

Reply:
Also FWIW we do cover `tidyr::separate()` in the notebook before this, but I like the idea of using it here, too, with the caveat that this may not always work like you want.

Reply:
I've talked myself out of this _because_ just because a single input gene set is used doesn't mean that that's the only association. So instead, I'm adding a note to always check `summary`.

Reply:
🎢